src/test/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookConsumerTest.java

```diff
@@ -160,10 +256,15 @@ public void testPullRequestCloseDoesntTriggerBitbucketSCMBuild() {
         verify(workflowTrigger, never())
                 .trigger(
                         eq(BitbucketWebhookTriggerRequest.builder().actor(BITBUCKET_USER).build()));
+
+        SCMHeadEvent<PullRequestClosedWebhookEvent> event = events.poll(500L, TimeUnit.MILLISECONDS);
+        assertThat(event.getPayload(), equalTo(pullRequestClosedEvent));
```

Comment:
this will throw an NPE if the `poll` times out and returns `null`. I think we should have a `assertThat(event, is(notNullValue())` above this line, and everywhere we get an event off the list via a `poll`